bottlerocket/types/src/agent_config.rs

```diff
@@ -495,12 +495,25 @@ fn k8s_version_valid() {
 pub struct WorkloadTest {
     pub name: String,
     pub image: String,
+    #[serde(default)]
+    pub gpu: bool,
 }
 
 #[derive(Debug, Clone, Default, Serialize, Deserialize, Configuration, Builder)]
 #[serde(rename_all = "camelCase")]
 #[crd("Test")]
 pub struct WorkloadConfig {
-    pub kubeconfig_base64: String,
-    pub plugins: Vec<WorkloadTest>,
+    pub kubeconfig_base64: Option<String>,
+    pub tests: Vec<WorkloadTest>,
+}
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize, Configuration, Builder)]
+#[serde(rename_all = "camelCase")]
+#[crd("Test")]
+pub struct EcsWorkloadTestConfig {
+    pub region: Option<String>,
+    pub cluster_name: String,
+    /// The role that should be assumed for this test agent.
+    pub assume_role: Option<String>,
```

Comment:
We probably need an assumed role for the k8s workload too.

Reply:
Let's leave that for a separate PR. With the SDK client used in the ECS testing it's an easy matter of just passing this in to the client initialization. I think the EKS/K8s code will need a little more work to enable that.

Reply:
Filed https://github.com/bottlerocket-os/bottlerocket-test-system/issues/732 so we don't lose that.